internal/charmstore/store.go

```diff
@@ -526,9 +526,16 @@ func (s *Store) AddCharm(c charm.Charm, p AddParams) (err error) {
 	if p.URL.URL.Series == "bundle" || p.URL.URL.User == "" || p.URL.URL.Revision == -1 {
```

Comment:
how about:

    id := &p.URL.URL

then you can get rid of many occurrences of p.URL.URL below ?